R/credentials.R

```diff
@@ -48,6 +48,15 @@ make_cred_cb <- function(password = askpass, verbose = TRUE){
       if(nchar(github_pat) > 0 && grepl('^https?://([^/]*@)?github.com', url)){
         return(c("git", github_pat))
       }
+
+      # Try looking for (cached) gitcreds, mainly to support GHE
+      # This should never cause a prompt! It either returns creds or errors.
+      if(requireNamespace('gitcreds')){
+        creds <- tryCatch(gitcreds::gitcreds_get(url, set_cache = FALSE), error = function(e){})
```

Comment:
Seems like allowing `set_cache = TRUE` (the default) would be OK or maybe even desirable?

Reply:
I'm a bit worried about saving passwords found in the store as plain-text envvars, as a side effect of pushing to git. My intention was just to use the `GITHUB_PAT_BERKELEY_EDU` variable if it exists.